src/util/dynamicfilesequencedownloader.cpp

```diff
@@ -0,0 +1,636 @@
+    response.start();
+    response.wait();
+
+    bool success = false;
+    int attempt = 0;
+    constexpr int MaxRetries = 1;
+    nlohmann::json jsonResult;
+
+    /********************
+    *   Example response
+    *********************
+    * {
+    *  "dataID": 1234,
+    *  "files": [
+    *   {
+    *    "timestamp": "2017-07-01 00:42:02.0",
+    *    "url": "https://iswa...fieldlines/trace_scs_outtoin/timestamp.osfls"
+    *   },
+    *   {
+    *    "timestamp": "2017-07-01 00:51:36.0",
+    *    "url": "https://iswa...fieldlines/trace_scs_outtoin/timestamp.osfls"
+    *   }
+    *  ],
+    *  "time.max": "2017-10-01 00:00:00.0",
+    *  "time.min": "2017-06-01 00:00:00.0"
+    * }
+    ********************
+    * note that requested time can be month 10 but last entry in list is month 07,
+    * meaning there are no more available files between month 7-10.
+    * *****************/
+    while (attempt <= MaxRetries && !success) {
+        try {
+            std::vector<char> data = response.downloadedData();
+            if (data.empty()) {
+                throw ghoul::RuntimeError("Empty HTTP response");
+            }
+            // @TODO (2025-06-10, Elon) What value is actually too large to handle?
+            if (data.size() > std::numeric_limits<std::size_t>::max()) {
+                throw ghoul::RuntimeError(
+                    "Http response with list of available files is too large, i.e. too many files"
+                );
+            }
+
+            jsonResult = nlohmann::json::parse(data);
+            success = true;
+        }
+        catch (const nlohmann::json::parse_error& ex) {
+            LERROR(std::format("JSON parsing error: '{}'", ex.what()));
+        }
+        catch (const ghoul::RuntimeError& ex) {
+            LERROR(std::format("An error occurred: '{}'", ex.what()));
+        }
+
+        if (!success) {
+            if (attempt < MaxRetries) {
+                LINFO(std::format("Retry nr {}.", attempt + 1));
+                std::this_thread::sleep_for(std::chrono::seconds(1));
+
+                response.start();
+                response.wait();
+            }
+            else {
+                LERROR(std::format(
+                    "Failed according to warning above with HTTP request of URL: {}",
+                    httpDataRequest
+                ));
+            }
+        }
+        attempt++;
+    }
+
+    if (!success) {
+        return;
+    }
+
+    int index = 0;
+    for (const nlohmann::json& element : jsonResult["files"]) {
+        std::string timestamp = element["timestamp"].get<std::string>();
+        std::string url = element["url"].get<std::string>();
+
+        //timestamp = "2022-11-13T16:14:00.000";
+        //url =
+        //https://iswaA-webservice1.ccmc.gsfc.nasa.gov/
+        //...iswa_data_tree/model/solar/WSA5.X/fieldlines/
+        //...GONG_Z/trace_pfss_intoout/2022/11/2022-11-13T16-14-00.000.osfls";
```

Comment:
What is this? An exmaple of how the data looks like?  Add another line to this comment that explains what this is , and some spacing after the `//` for readability
```suggestion
        // <Add an explanation here>
        // timestamp = "2022-11-13T16:14:00.000";
        // url =
        // https://iswaA-webservice1.ccmc.gsfc.nasa.gov/
        // ...iswa_data_tree/model/solar/WSA5.X/fieldlines/
        // ...GONG_Z/trace_pfss_intoout/2022/11/2022-11-13T16-14-00.000.osfls";
```